container.go

```diff
@@ -57,41 +57,9 @@ func New() Container {
 	}
 }
 
-// Connect a component, optionally identified by id.
-func (container Container) Connect(val interface{}, id ...string) {
-	ptr := false
-	rv := reflect.ValueOf(val)
-	rt := rv.Type()
-	nam := ""
-	_, file, no, _ := runtime.Caller(container.callerSkip + 1)
-
-	if len(id) > 0 {
-		nam = id[0]
-	}
-
-	comp := component{
-		id:         nam,
-		value:      rv,
-		declaredAt: file + ":" + strconv.Itoa(no),
-	}
-
-	if rt.Kind() == reflect.Ptr {
-		rt = rt.Elem()
-		rv = rv.Elem()
-		comp.value = rv
-		ptr = true
-	}
-
-	if gr, ok := container.components[rt]; ok {
-		if comp, ok := gr.find(nam); ok {
-			panic(duplicateError{previous: comp})
-		}
-	}
+func dependinces(container Container, rt reflect.Type, rv reflect.Value) []dependency {
 
-	if rt.Kind() != reflect.Struct {
-		container.components[rt] = append(container.components[rt], comp)
-		return
-	}
+	dependinces := make([]dependency, 0)
```

Comment:
isn't this supposed to be `dependencies`?
https://www.collinsdictionary.com/dictionary/english/dependencies